test/elixir_analyzer/exercise_test/common_checks/variable_names_test.exs

```diff
@@ -0,0 +1,419 @@
+# credo:disable-for-this-file Credo.Check.Readability.VariableNames
+
+defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.VariableNamesTest do
+  use ExUnit.Case
+  alias ElixirAnalyzer.Comment
+  alias ElixirAnalyzer.Constants
+  alias ElixirAnalyzer.ExerciseTest.CommonChecks.VariableNames
+
+  @moduledoc """
+  Tests that variables are in snake_case
+  tests taken and adapted from
+  https://github.com/rrrene/credo/blob/master/test/credo/check/readability/variable_names_test.exs
+  """
+
+  describe "valid variable names" do
+    test "it should NOT report expected code" do
+      code =
+        quote do
+          defmodule CredoSampleModule do
+            def some_function(parameter1, parameter2) do
+              some_value = parameter1 + parameter2
+              {some_value, _} = parameter1
+              [1, some_value] = parameter1
+              [some_value | tail] = parameter1
+              "e" <> some_value = parameter1
+              ^some_value = parameter1
+              %{some_value: some_value} = parameter1
+              #    ... = parameter1 # Ignore this pathological case
+              latency_μs = 5
```

Comment:
no, wait, here we go, more cases that need to pass the test 😇 see https://github.com/rrrene/credo/blob/master/lib/credo/check/readability/variable_names.ex#L24
```suggestion
              latency_μs = 5
              user = %__MODULE__{}
              IO.inspect(__DIR__)
              IO.inspect(__ENV__)
            end

            defmacro foo() do
              IO.inspect(__CALLER__)
            end
```

Reply:
Oh, nice catch, thank you!

Can I ask an unrelated question? I've been thinking about going into software/web development. If this PR was a professional one, how would it be received? Would you feel that I'm not doing my job right? Would you say it's OK from a junior developer only, or is this standard and expected that things like this will be missed? This is certainly showing me the necessity of peer review (and also that tests are hard).

Reply:
I'll reply to this question on Slack, it feels a bit too personal to answer publicly 😅